src/components/SmartInput/index.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import Select, {
+  MultiValue,
+  Props,
+  GroupBase,
+  SelectInstance,
+  Options,
+} from 'react-select';
+import { customComponents, customStyles, customThemes } from './CustomSelectUtils';
+import DropdownOption from './CustomSelectComponents/DropdownOption';
+
+declare module 'react-select/dist/declarations/src/Select' {
+  export interface Props<
+    Option,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    IsMulti extends boolean,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Is this necessary?

Reply:
yes, i need to "augment" this type, and eslint complains about Generics not being used, but if I remove them it stops working.

Reply:
I guess my question is that if you're augmenting it, then it's not unused right? Is there a way to do this without needing the ignore? Maybe additional configuration?

Reply:
https://react-select.com/typescript#custom-select-props its in here, it doesn't give too much info.

but if i remove it, it says that all declarations must be identical
<img width="771" alt="image" src="https://user-images.githubusercontent.com/69323251/160490613-6275930e-2fa9-4639-ad8f-d0284db127cb.png">

doing more research on it.

Reply:
i can do something like this
<img width="585" alt="image" src="https://user-images.githubusercontent.com/69323251/160495071-665899b0-81aa-4501-83dd-865e58e4f9bb.png">

but i honestly don't like it because `formatGroupLabel` is only being put in there again to stop linter from complaining.

Reply:
@cbedwinotalvaro Any thoughts here?

Reply:
@reywright we discuss this in a call, him and David are both on board with disabling eslint rule and keeping a comment explaining why.